docs/docs/tutorials/jwt_authentication.md

```diff
@@ -0,0 +1,517 @@
+
+You should now have a directory called `authenticated_app`. Let's change directories into the newly created project:
+
+```bash
+cd authenticated_app
+```
+
+Then, run the following command:
+
+```bash
+dart_frog dev
+```
+
+This will start the development server on port `8080`:
+
+```
+✓ Running on http://localhost:8080 (1.3s)
+The Dart VM service is listening on http://127.0.0.1:8181/YKEF_nbwOpM=/
+The Dart DevTools debugger and profiler is available at: http://127.0.0.1:8181/YKEF_nbwOpM=/devtools/#/?uri=ws%3A%2F%2F127.0.0.1%3A8181%2FYKEF_nbwOpM%3D%2Fws
+[hotreload] Hot reload is enabled.
+```
+
+Make sure it's working by opening [http://localhost:8080](http://localhost:8080) in your browser or via `cURL`:
+
+```bash
+curl --request GET \
+  --url http://localhost:8080
+```
+
+If everything succeeded, you should see `Welcome to Dart Frog!`.
+
+## The hardcode domain code
+
+To keep the tutorial simple and focused on authentication, our database of users will be hardcoded and
+the `User` model will be simple, containing just an id and a name.
+
+For next steps, use the code below to create the domain classes.
+
+```dart
+// lib/user.dart
+class User {
+  const User({
+    required this.id,
+    required this.name,
+    required this.password,
+  });
+
+  final String id;
+  final String name;
+  final String password;
+}
+import 'package:authenticated_app/user.dart';
+```
+
+```dart
+// lib/authenticator.dart
+class Authenticator {
+  static const _users = {
+    'john': User(
+      id: '1',
+      name: 'John',
+      password: '123',
+    ),
+    'jack': User(
+      id: '2',
+      name: 'Jack',
+      password: '321',
+    ),
+  };
+
+  static const _passwords = {
+    // ⚠️ Never store user's password in plain text, these values are in plain text
+    // just for the sake of the tutorial.
+    'john': '123',
+    'jack': '321',
+  };
+
+  User? findByUsernameAndPassword({
+    required String username,
+    required String password,
+  }) {
+    final user = _users[username];
+
+    if (user != null && _passwords[username] == password) {
+      return user;
+    }
+
+    return null;
+  }
+}
+
+```
+
+We also need to provide our `Authenticator` to our routes. It will be used by the sign in route and all others that
+require authentication.
+
+In order to do so, we can use [Dart frog's dependency injection](https://dartfrog.vgv.dev/docs/basics/dependency-injection)
+and create a middleware in the root of our `routes` folder with the following code:
+
+```dart
+// lib/_middleware.dart
+import 'package:authenticated_app/authenticator.dart';
+import 'package:dart_frog/dart_frog.dart';
+
+Handler middleware(Handler handler) {
+  return handler.use(
+    provider<Authenticator>(
+      (_) => Authenticator(),
+    ),
+  );
+}
+```
+
+:::tip
+Install and use the [Dart Frog VS Code extension](https://marketplace.visualstudio.com/items?itemName=VeryGoodVentures.dart-frog) to easily create new middleware within your IDE.
```

Comment:
Cant we have a single tip that mentions it all?

Reply:
@alestiago this was your suggestion haha, argue for it

Reply:
My reasoning is this: its a good tip but imho should be right after the intro to explain that it makes everything easier (if u use vscode) but it does not add "tip" value to the related content.

As a developer I rather see tips for the content I am reading then 3 or 4 times an ad for a different product

Reply:
Thanks @erickzanardo for tagging me here. I think @wolfenrain is raising a valid concern, but I do have a slight preference to the current approach. I'll chat with @wolfenrain off Git-Hub to see what action we should take here 😝 . 

Personally, I think we can resolve this conversation if it is blocking this Pull Request. Committing to adding an intro that mentions everything can be made easier with the extension (as suggested by @wolfenrain), will impact all other tutorials since they currently follow the the explicit tip-approach. This means that, for consistency, we'll have to update all other tutorials and I think that can be a standalone change to avoid polluting this Pull Request.